components/server/src/projects/projects-service.ts

```diff
@@ -185,8 +185,8 @@ export class ProjectsService {
         if (!project) {
             throw new Error("Project not found");
         }
-        const normalizedContextUrl = this.contextParser.normalizeContextURL(project.cloneUrl);
-        const commitContext = (await this.contextParser.handle(ctx, user, normalizedContextUrl)) as CommitContext;
+
+        const commitContext = (await this.contextParser.handle(ctx, user, project.cloneUrl)) as CommitContext;
```

Comment:
I would revert this change. The contract of `contextParser.handle` is that it receives the output of `normalizedContextURL` and we should keep it that way.

I'm aware that this is neither well documented nor obvious from the interface (due to how it evolved over time). :wink:

Reply:
Sure, I just want to point out that looking at the code calling `normalizeContextURL` here does nothing as the url is invalid and the prefix needs to be removed first, which is what the `handle` method does internally

Reply:
> which is what the handle method does internally

Yes, unfortunately `normalize` is not idempotent. E.g., if you call it 2 times, it might have a different result then calling it 1 or 3 times in a row (as I mentionend: bad contract which evolved and should be refactored :100: )